src/core/Mixer.cpp

```diff
@@ -575,21 +575,35 @@ void Mixer::changeQuality( const struct qualitySettings & _qs )
 
 
 
-void Mixer::setAudioDevice( AudioDevice * _dev,
-			    bool startNow )
+void Mixer::doSetAudioDevice( AudioDevice * _dev )
 {
-	stopProcessing();
+	// TODO: Use shared_ptr here in the future.
```

Comment:
Why don't you do this now? Also, `QSharedPointer` will work as well as `std::shared_pointer`.

Reply:
Fixed in fc5cb9c